### ISSUE TITLE
Additional animation issues

### ISSUE THREAD

**Issue by rreusser** [plotly/plotly.js#933]
I've found some additional issues related to animation that should be fixed ASAP:
- [x] transforms are an array, which extendDeepNoArrays accidentally avoids extending
- [ ] animate doesn't obey the correct semantics where changes are applied to all traces when trace indices not specified (copy from generalized function as a result of `Plotly.update`?)
- [ ] animate should be able to apply more than one nextFrame per frame in order to keep up with changes
- [x] animation should emit an event on each frame _so that you can figure out which frame it's on_! This is crucial for displaying information about an ongoing animation.
- [ ] convert async sequences to `Lib.syncOrAsync` (see first try in https://github.com/plotly/plotly.js/pull/1693)

**Comment by rreusser** [comment]
@etpinard @monfera Curious to hear your thoughts on the merging issue.

Animations currently expand objects then `extendDeepNoArrays`. That's mostly okay except for a couple issues:

- Transforms don't get properly merged because of the NoArrays part
- `{'marker.color': ...}` and `marker: {color: ...}` both have the same effect, which makes frames nice and easy to write, but is slightly different than the rest of plotly which overwrites all marker properties for the second case.
- animations don't do array indexing like `{'transforms.0.type': ...}`

Long story short, I need to fix a couple bugs, so I either need to commit to the current approach and make it 100% correct, or I need to reevaluate and make the right choice about the most straightforward and sensible way for this to work.

**Comment by rreusser** [comment]
After a brief discussion, the conclusion is that the current behavior is best: attribute strings are good for single properties but awkward for whole frames (which would then need *every* property attribute-stringed, which wouldn't really be acceptable).

Then working through transform array logic…

**Comment by etpinard** [comment]
After https://github.com/plotly/plotly.js/pull/956 it would be easy

- [] add animation for `scatter` trace with `'text'` mode.

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
